src/Relation/Binary/Reasoning/Base/Single.agda

```diff
@@ -13,17 +13,59 @@ module Relation.Binary.Reasoning.Base.Single
   (refl : Reflexive _∼_) (trans : Transitive _∼_)
   where
 
+open import Level using (_⊔_)
+import Relation.Binary.PropositionalEquality.Core as P
+open P using (_≡_)
+
+infix  4 _IsRelatedTo_
+infix  3 _∎
+infixr 2 step-∼ step-≡ step-≡˘
+infixr 2 _≡⟨⟩_
+infix  1 begin_
+
 ------------------------------------------------------------------------
 -- Reasoning combinators
 
 -- Re-export combinators from partial reasoning
 
-open import Relation.Binary.Reasoning.Base.Partial _∼_ trans public
-  hiding (_∎⟨_⟩)
+data _IsRelatedTo_ : A → A → Set (a ⊔ ℓ) where
+  emp   : ∀ x → x IsRelatedTo x
```

Comment:
Hmm sorry just re-read your original issue and I missed the bit about changing setoid reasoning as well. I agree that in the case of partial setoids, having to provide the unnecessary refl proof was problematic as a) extra typing for the user and b) the relation may not even be reflexive. 

However in the single/setoid case, we do have access to the `refl` proof and therefore the user never needs to explicitly provide it. Even if using it is technically unnecessary, is there a compelling reason for why we should break backwards compatibility to avoid using it?

Even if there is, I would be more tempted to put off changing it until v2.0 and then change all the reasoning APIs at once. After all the sample problem happens with poset/preorder reasoning right? Changing those may be more complicated...

Reply:
does it actually break anything if we omit an unnecessary proof? the fact that people doesn't complain about an extra refl proof seems to indicate that usually people just don't care.

we can then postpone this PR to 2.0 if you find it better this way. any workaround would complicate the PR and introduce later technical debt to pay back.

Reply:
> does it actually break anything if we omit an unnecessary proof? 

Well adding the extra `emp` constructor "unnecessarily" breaks the definition of any reasoning API that a user has built on top of `Relation.Binary.Reasoning.Base.Single` right? For example in the standard library it breaks `Reasoning.MultiSetoid` and `PermutationReasoning`. Note, it's not breaking the use of such reasoning APIs, only their definition.

> we can then postpone this PR to 2.0 if you find it better this way. any workaround would complicate the PR and introduce later technical debt to pay back.

Well ideally I'd like to keep the changes to `Relation.Binary.Reasoning.Base.Partial` and merge them in v1.6 as there is a clear advantage to incorporating them now as it directly simplifies the user-facing API. 

In contrast any changes to eliminate the redundant reflexive cases in `Relation.Binary.Reasoning.Base.(Single/Double/Triple)` do not simplify the user-facing API and therefore if we are to make them, they should be made consistently across all three of those modules and we should wait until v2.0 to do it? Does that make sense?